workspaces/arborist/test/arborist/rebuild.js

```diff
@@ -13,16 +13,15 @@ const fixture = (t, p) => require(`${fixtures}/reify-cases/${p}`)(t)
 const isWindows = process.platform === 'win32'
 const PORT = 12345 + (+process.env.TAP_CHILD_ID || 0)
 
-const server = require('http').createServer(() => {
+const server = require('node:http').createServer(() => {
   throw new Error('rebuild should not hit the registry')
 })
 t.before(() => new Promise(res => {
-  server.listen(PORT, () => {
-    t.teardown(() => server.close())
-    res()
-  })
+  server.listen(PORT, res)
 }))
 
+t.teardown(() => server.close())
```

Comment:
I believe this was done as a defensive measure in case `listen` does not succeed. We don't need `close` to run then, and we'd potentially get extra errors if we tried to close a server that never finished listening.